lib/screens/search/search_app_textfield.dart

```diff
@@ -1,86 +1,130 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
 import 'package:guam_community_client/styles/colors.dart';
 import 'package:provider/provider.dart';
 import '../../commons/common_text_button.dart';
 import 'package:guam_community_client/styles/fonts.dart';
 import '../../providers/search/search.dart';
-import '../../providers/user_auth/authenticate.dart';
 
 class SearchAppTextField extends StatefulWidget {
+  final bool showHistory;
+  final Function showSearchHistory;
+
+  SearchAppTextField(this.showHistory, this.showSearchHistory);
+
   @override
   State<StatefulWidget> createState() => SearchAppTextFieldState();
 }
 
 class SearchAppTextFieldState extends State<SearchAppTextField> {
   static final controller = TextEditingController();
+  bool isTextEmpty = true;
+
+  @override
+  void initState() {
+    controller.text = '';
+    super.initState();
+  }
 
   @override
   void dispose() {
     controller.dispose();
     super.dispose();
   }
 
+  void checkTextEmpty(e) {
+    setState(() => e == ''
+        ? isTextEmpty = true
+        : isTextEmpty = false || widget.showHistory
```

Comment:
Can't understand the semantic of 
`isTextEmpty = (false || widget.showHistory)`